src/main/java/org/alfresco/repo/security/permissions/impl/acegi/ACLEntryAfterInvocationProvider.java

```diff
@@ -573,244 +554,194 @@ private ResultSet decideNew(Authentication authentication, Object object, Config
-        if (returnedObject.getResultSetMetaData().getSearchParameters().getMaxPermissionCheckTimeMillis() >= 0)
+    private ResultSet decideNew(Authentication authentication, Object object, ConfigAttributeDefinition config, ResultSet returnedObject) throws AccessDeniedException
+    {
+        if (returnedObject == null)
         {
-            maxCheckTime = returnedObject.getResultSetMetaData().getSearchParameters().getMaxPermissionCheckTimeMillis();
+            return null;
         }
 
-        if (supportedDefinitions.size() == 0)
+        Integer maxSize = getMaxSize(returnedObject);
+
+        // Create result set
+        ResultSet resultSet = decidePermissions(returnedObject, null);
+        if (maxSize == null)
         {
-            if (maxSize == null)
-            {
-                return returnedObject;
-            }
-            else if (returnedObject.length() > maxSize.intValue())
-            {
-                for (int i = 0; i < maxSize.intValue(); i++)
-                {
-                    filteringResultSet.setIncluded(i, true);
-                }
-                filteringResultSet.setResultSetMetaData(new SimpleResultSetMetaData(LimitBy.FINAL_SIZE, PermissionEvaluationMode.EAGER, returnedObject.getResultSetMetaData()
-                        .getSearchParameters()));
-                return filteringResultSet;
-            }
-            else
-            {
-                for (int i = 0; i < returnedObject.length(); i++)
-                {
-                    filteringResultSet.setIncluded(i, true);
-                }
-                filteringResultSet.setResultSetMetaData(new SimpleResultSetMetaData(returnedObject.getResultSetMetaData().getLimitedBy(), PermissionEvaluationMode.EAGER, returnedObject.getResultSetMetaData()
-                        .getSearchParameters()));
-                return filteringResultSet;
-            }
-            
-        }
-        if (returnedObject.length() > 0)
-        {
-            // force prefetch before starting record time
-            boolean builkFetch = returnedObject.getBulkFetch();
-            returnedObject.setBulkFetch(false);
-            returnedObject.getNodeRef(returnedObject.length() - 1);
-            returnedObject.setBulkFetch(builkFetch);
+            return resultSet;
         }
 
-        // record the start time
-        long startTimeMillis = System.currentTimeMillis();
-        // set the default, unlimited resultset type
-        filteringResultSet.setResultSetMetaData(new SimpleResultSetMetaData(returnedObject.getResultSetMetaData().getLimitedBy(), PermissionEvaluationMode.EAGER, returnedObject.getResultSetMetaData()
-                .getSearchParameters()));
+        return filterMaxCount(maxSize, resultSet);
+    }
 
-        for (int i = 0; i < returnedObject.length(); i++)
+    private ResultSet decideOld(Authentication authentication, Object object, ConfigAttributeDefinition config, ResultSet returnedObject) throws AccessDeniedException

```

Comment:
Javadoc.  What does "decideOld" mean compared with "decideNew"?  I can't think of anything that should be old and new here.